src/components/Account.tsx

```diff
@@ -0,0 +1,170 @@
+        .from('profiles')
+        .select(`first_name, last_name, birthday, gender, race_ethnicity`)
+        .eq('user_id', session?.user.id)
+        .single();
+
+      if (error && status !== 406) {
+        throw error;
+      }
+
+      if (data) {
+        setFirstName(data.first_name || firstName);
+        setLastName(data.last_name || lastName);
+        setBirthday(new Date(data.birthday) || birthday);
+        setGender(data.gender || gender);
+        setRaceEthnicity(data.race_ethnicity || raceEthnicity);
+      }
+    } catch (error) {
+      if (error instanceof Error) {
+        Alert.alert(`Get profile error: ${error.message}`);
+      }
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    if (session) getProfile();
+  }, [session]);
+
+  const updateProfile = async () => {
+    try {
+      setLoading(true);
+      if (!session?.user) throw new Error('No user on the session!');
+
+      // Only update value that are not blank
+      const updates = {
+        ...(firstName && { first_name: firstName }),
+        ...(lastName && { last_name: lastName }),
+        ...(gender && { gender }),
+        ...(raceEthnicity && { race_ethnicity: raceEthnicity }),
+        ...(birthday && { birthday }),
+      };
+
+      // Check if user exists
+      const { count } = await supabase
+        .from('profiles')
+        .select(`*`, { count: 'exact' })
+        .eq('user_id', session?.user.id);
+
+      if (count && count >= 1) {
+        // Update user if they exist
+        const { error } = await supabase
+          .from('profiles')
+          .update(updates)
+          .eq('user_id', session?.user.id)
+          .select('*');
+
+        if (error) throw error;
+      } else {
+        // Create user if they don't exist
+        const { error } = await supabase.from('profiles').insert(updates);
+
+        if (error) throw error;
+      }
+
+      Alert.alert('Succesfully updated user!');
+    } catch (error) {
+      if (error instanceof Error) {
+        Alert.alert(error.message);
+      }
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  return (
+    <ScrollView style={styles.container}>
+      <View style={[styles.verticallySpaced, styles.mt20]}>
+        <Input label="Email" value={session?.user?.email} disabled />
+      </View>
+
+      <UserStringInput
+        label="First Name"
+        value={firstName}
+        set={setFirstName}
+      />
+      <UserStringInput label="Last Name" value={lastName} set={setLastName} />
+      <UserStringInput label="Gender" value={gender} set={setGender} />
+      <UserStringInput
+        label="Race/Ethnicity"
+        value={raceEthnicity}
+        set={setRaceEthnicity}
+      />
+
+      <DateTimePicker
+        testID="dateTimePicker"
+        value={birthday}
+        mode="date"
+        onChange={date => {
+          setBirthday(new Date(date.nativeEvent.timestamp));
+        }}
+      />
```

Comment:
When testing on Android and I land on the Account screen, the datetimepicker modal is already open and can't be exited out of to update any of the other profile details. I would look at "Android imperative api" section of the [docs for the package](https://www.npmjs.com/package/@react-native-community/datetimepicker#android-imperative-api), which mentions that the declarative component api is more prone to bugs.